tests/rkt_config_test.go

```diff
@@ -0,0 +1,483 @@
+							basic: &basic{"docker", "dockerPassword"},
+						},
+					},
+				},
+				{
+					RktVersion: "v1",
+					RktKind:    "dockerAuth",
+					auth: &auth{
+						Registries: []string{"gcr.io"},
+						Credentials: credentials{
+							basic: &basic{"docker", "dockerPassword"},
+						},
+					},
+				},
+			},
+		},
+
+		{ // overwrite one auth domain, one dockerAuth registry in user dir
+			configFunc: func(ctx *testutils.RktRunCtx) {
+				writeConfig(t, authDir(ctx.SystemDir()), "basic.json", mustMarshalJSON(
+					cfg{
+						RktVersion: "v1",
+						RktKind:    "auth",
+						auth: &auth{
+							Type:    "basic",
+							Domains: []string{"tectonic.com", "coreos.com"},
+							Credentials: credentials{
+								basic: &basic{"user", "userPassword"},
+							},
+						},
+					},
+				))
+
+				writeConfig(t, authDir(ctx.UserDir()), "basic.json", mustMarshalJSON(
+					cfg{
+						RktVersion: "v1",
+						RktKind:    "auth",
+						auth: &auth{
+							Type:    "oauth",
+							Domains: []string{"tectonic.com"},
+							Credentials: credentials{
+								oauth: &oauth{"someToken"},
+							},
+						},
+					},
+				))
+
+				writeConfig(t, authDir(ctx.SystemDir()), "docker.json", mustMarshalJSON(
+					cfg{
+						RktVersion: "v1",
+						RktKind:    "dockerAuth",
+						auth: &auth{
+							Registries: []string{"quay.io", "gcr.io"},
+							Credentials: credentials{
+								basic: &basic{"docker", "dockerPassword"},
+							},
+						},
+					},
+				))
+
+				writeConfig(t, authDir(ctx.UserDir()), "docker.json", mustMarshalJSON(
+					cfg{
+						RktVersion: "v1",
+						RktKind:    "dockerAuth",
+						auth: &auth{
+							Registries: []string{"gcr.io"},
+							Credentials: credentials{
+								basic: &basic{"over", "written"},
+							},
+						},
+					},
+				))
+			},
+			expected: []cfg{
+				{
+					RktVersion: "v1",
+					RktKind:    "auth",
+					auth: &auth{
+						Type:    "oauth",
+						Domains: []string{"tectonic.com"},
+						Credentials: credentials{
+							oauth: &oauth{"someToken"},
+						},
+					},
+				},
+				{
+					RktVersion: "v1",
+					RktKind:    "auth",
+					auth: &auth{
+						Type:    "basic",
+						Domains: []string{"coreos.com"},
+						Credentials: credentials{
+							basic: &basic{"user", "userPassword"},
+						},
+					},
+				},
+				{
+					RktVersion: "v1",
+					RktKind:    "dockerAuth",
+					auth: &auth{
+						Registries: []string{"quay.io"},
+						Credentials: credentials{
+							basic: &basic{"docker", "dockerPassword"},
+						},
+					},
+				},
+				{
+					RktVersion: "v1",
+					RktKind:    "dockerAuth",
+					auth: &auth{
+						Registries: []string{"gcr.io"},
+						Credentials: credentials{
+							basic: &basic{"over", "written"},
+						},
+					},
+				},
+			},
+		},
+	} {
+		func() {
+			ctx := testutils.NewRktRunCtx()
+			defer ctx.Cleanup()
+
+			tt.configFunc(ctx)
+
+			rktCmd := ctx.Cmd() + " --debug --insecure-options=image config"
```

Comment:
We are using the `--debug` option in pretty much every other test. I suggest to open a different issue such that we evaluate the removal for all tests then.